src/Data/Impact.elm

```diff
@@ -153,7 +153,18 @@ noStepsImpacts =
     }
 
 
-stepsColors : { materials : String, transform : String, packaging : String, transports : String, distribution : String, usage : String, endOfLife : String }
+type alias StepsColors =
+    { materials : String
+    , transform : String
+    , packaging : String
+    , transports : String
+    , distribution : String
+    , usage : String
+    , endOfLife : String
+    }
```

Comment:
Now if you look at the `StepImpacts`, it feels like we could abstract the type to accept a type variable.

Reply:
Something like this:

```diff
diff --git a/src/Data/Impact.elm b/src/Data/Impact.elm
index 5dd8d9fa..2a3b5742 100644
--- a/src/Data/Impact.elm
+++ b/src/Data/Impact.elm
@@ -130,17 +130,21 @@ totalComplementsImpactAsChartEntry complementsImpacts =
 -- Steps impacts
 
 
-type alias StepsImpacts =
-    { materials : Maybe Unit.Impact
-    , transform : Maybe Unit.Impact
-    , packaging : Maybe Unit.Impact
-    , transports : Maybe Unit.Impact
-    , distribution : Maybe Unit.Impact
-    , usage : Maybe Unit.Impact
-    , endOfLife : Maybe Unit.Impact
+type alias Steps a =
+    { materials : a
+    , transform : a
+    , packaging : a
+    , transports : a
+    , distribution : a
+    , usage : a
+    , endOfLife : a
     }
 
 
+type alias StepsImpacts =
+    Steps (Maybe Unit.Impact)
+
+
 noStepsImpacts : StepsImpacts
 noStepsImpacts =
     { materials = Nothing
@@ -154,14 +158,7 @@ noStepsImpacts =
 
 
 type alias StepsColors =
-    { materials : String
-    , transform : String
-    , packaging : String
-    , transports : String
-    , distribution : String
-    , usage : String
-    , endOfLife : String
-    }
+    Steps String
 
 
 stepsColors : StepsColors
```

Reply:
Yeah, makes sense, adding this, then merging. Thanks!